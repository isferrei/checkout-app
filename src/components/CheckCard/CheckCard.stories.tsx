import type { Meta, StoryObj } from "@storybook/react";

import { CheckCard } from "./CheckCard";

const meta = {
  title: "Example/CheckCard",
  component: CheckCard,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta;

export default meta;
type Story = StoryObj;

export const Default = {
  args: {
    id: "1",
    title: "Anual",
    description: "À Vista",
    fullPrice: "514",
    discountAmmount: "10",
    installments: "12",
    discountPercentage: "0.1",
    checked: "true",
    periodLabel: "mês",
    label: "Anual  |  À Vista",
    price: "De R$ 514,80  |  Por R$ 436,90",
    discount: "-15%",
    value: "-15%",
    onChange: () => {},
  },
} satisfies Story;
