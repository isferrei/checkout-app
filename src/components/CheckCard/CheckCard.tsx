import { useState } from "react";

type CheckCardProps = {
  label?: string;
  price: string;
  discount: string;
  description: string;
  checked: boolean;
  onChange: (newValue: string) => void;
  value: string;
};

export function CheckCard(props: CheckCardProps) {
  const [isChecked, setIsChecked] = useState(props.checked);

  const handleOnChange = (event: { target: { checked: any } }) => {
    const newValue = event.target.checked;
    setIsChecked(newValue);
    if (props.onChange) {
      props.onChange(newValue);
    }
  };

  return (
    <label
      htmlFor="radio"
      className="border-[1px] flex justify-between flex-row font-sans items-center w-[20rem] border-blue rounded-[0.93rem] p-[1.25rem] cursor-pointer focus:outline-none focus:ring-2 hover:border-2 transition-all"
    >
      <div className="flex gap-[0.75rem] flex-row justify-center items-center">
        <section className="text-blue font-normal">
          <h2 className="font-bold text-[0.87rem]">{props.label}</h2>
          <p className="text-[0.75rem]">{props.price}</p>
          <p className="text-[0.65rem] text-orange">{props.description}</p>
        </section>
        <section className="flex justify-center items-center h-[1rem] w-[2.5rem] px-[0.3rem] py-[0.25rem] text-white text-[0.63rem] rounded-[0.56rem] bg-orange">
          {props.discount}
        </section>
      </div>
      <label>
        <input
          type="radio"
          id="radio"
          value={props.value}
          checked={isChecked}
          onChange={handleOnChange}
          className="hidden"
        />
        <span className="w-[16px] h-[16px] rounded-full border-2 border-lightGray flex items-center justify-center">
          {isChecked && (
            <span className="w-[8px] h-[8px] rounded-full bg-blue"></span>
          )}
        </span>
      </label>
    </label>
  );
}
