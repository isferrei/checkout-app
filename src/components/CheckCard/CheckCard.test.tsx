import { render, screen, fireEvent } from "@testing-library/react";
import { CheckCard } from "./CheckCard";

describe("CheckCard", () => {
  const props = {
    id: "card1",
    title: "Premium Anual",
    description: "Parcelado",
    fullPrice: 100,
    discountAmount: 10,
    installments: 3,
    discountPercentage: 0.1,
    checked: false,
    periodLabel: "month",
    onChange: jest.fn(),
    value: "card1",
  };

  test("renders CheckCard component with correct props", () => {
    const { container } = render(<CheckCard discountAmmount={0} {...props} />);

    expect(screen.getByTestId("checkcard-title-description")).toHaveTextContent(
      "Premium Anual | Parcelado"
    );

    expect(
      screen.getByTestId("checkcard-full-price-discount")
    ).toHaveTextContent("De R$ 100,00 | Por R$ 100,00");

    expect(screen.getByTestId("checkcard-installments")).toHaveTextContent(
      "3x de R$ 33,33/month"
    );

    expect(screen.getByText("-10%")).toBeInTheDocument();

    fireEvent.click(screen.getByTestId("checkcard"));

    expect(props.onChange).toHaveBeenCalledTimes(1);
    expect(props.onChange).toHaveBeenCalledWith("card1");

    expect(container.firstChild).toMatchSnapshot();
  });
});
