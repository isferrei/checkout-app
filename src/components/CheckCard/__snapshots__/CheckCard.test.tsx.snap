// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckCard renders CheckCard component with correct props 1`] = `
<label
  class="border-[1px] flex justify-between flex-row font-sans items-center w-[20rem] border-blue rounded-[0.93rem] p-[1.25rem] cursor-pointer focus:outline-none focus:ring-2 hover:bg-[#F4F3F6] transition-all"
  data-testid="checkcard"
  for="card1"
>
  <div
    class="flex gap-[0.75rem] flex-row justify-center items-center"
  >
    <section
      class="text-blue font-normal"
    >
      <h2
        class="font-bold text-[0.87rem]"
        data-testid="checkcard-title-description"
      >
        Premium Anual
         | 
        Parcelado
      </h2>
      <p
        class="text-[0.75rem]"
        data-testid="checkcard-full-price-discount"
      >
        De R$ 
        100,00
         | Por R$
         
        100,00
      </p>
      <p
        class="text-[0.65rem] text-orange"
        data-testid="checkcard-installments"
      >
        3
        x de R$
         
        33,33
        /
        month
      </p>
    </section>
    <section
      class="flex justify-center items-center h-[1rem] w-[2.5rem] px-[0.3rem] py-[0.25rem] text-white text-[0.63rem] rounded-[0.56rem] bg-orange"
    >
      -10%
    </section>
  </div>
  <label>
    <input
      class="hidden"
      id="card1"
      name="plan_card"
      type="radio"
      value="card1"
    />
    <span
      class="w-[16px] h-[16px] rounded-full border-2 flex items-center justify-center transition-all"
      style="border-color: #c9c5d4;"
    >
      <span
        class="w-[8px] h-[8px] rounded-full bg-blue"
      />
    </span>
  </label>
</label>
`;
